src/index.js

```diff
@@ -398,6 +401,71 @@ Compute.prototype.createImage = function(name, disk, options, callback) {
   );
 };
 
+/**
+ * Create an instance template.
+ *
+ * @param {string} name - The name of the target image.
+ * @param {object=} options - See an
+ *     [InstanceTemplate resource](https://cloud.google.com/compute/docs/reference/v1/instanceTemplate#resource).
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {InstanceTemplate} callback.instanceTemplate - The created InstanceTemplate.
+ * @param {Operation} callback.operation - An operation object that can be used
+ *     to check the status of the request.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * const Compute = require('@google-cloud/compute');
+ * const compute = new Compute();
+ *
+ * function onCreated(err, instanceTemplate, operation, apiResponse) {
+ *   // `instanceTemplate` is an InstanceTemplate object.
+ *
+ *   // `operation` is an Operation object that can be used to check the
+ *   // status of the request.
+ * }
+ *
+ * compute.createInstanceTemplate('my-instance-template', options, onCreated);
+ *
+ * //-
+ * // If the callback is omitted, we'll return a Promise.
+ * //-
+ * compute.createInstanceTemplate('my-instance-template', options).then(function(data) {
+ *   const instanceTemplate = data[0];
+ *   const operation = data[1];
+ *   const apiResponse = data[2];
+ * });
+ */
+Compute.prototype.createInstanceTemplate = function(name, options, callback) {
```

Comment:
Just a convention, when a configuration object is required, we call it `config`. We save `options` for a configuration object that is optional.